plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDecimalCoercers.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.DecimalParseResult;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Decimals;
+import io.trino.spi.type.Int128;
+import io.trino.spi.type.Type;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.HiveType.toHiveType;
+import static io.trino.plugin.hive.coercions.CoercionUtils.createCoercer;
+import static io.trino.spi.predicate.Utils.blockToNativeValue;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDecimalCoercers
+{
+    private static final DecimalType LONG_DECIMAL_TYPE = DecimalType.createDecimalType(30, 12);
+    private static final DecimalType SHORT_DECIMAL_TYPE = DecimalType.createDecimalType(10, 5);
+
+    @Test(dataProvider = "dataProvider")
+    public void testDecimalToIntCoercion(String decimalString, Type coercedType, Object expectedValue)
+    {
+        DecimalParseResult parseResult = Decimals.parse(decimalString);
+        if (decimalString.length() > 19) {
```

Comment:
nit: We can also assert from the `parseResult#getType#isShort`

Reply:
Sorry I meant instead of relying on `decimalString.length` we could use `parseResult#getType#isShort`.

Btw what is the purpose of this assertion or can we remove it ?

Reply:
So I want to make sure that for all the cases we test both LongDecimals and ShortDecimals implementations. 
I cannot make this distinguish based on `isShort` because it is an effect not the cause